src/main/java/io/cdap/plugin/salesforce/SObjectFilterDescriptor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * SObject query filter values holder.
+ */
+public class SObjectFilterDescriptor {
+
+  private final ZonedDateTime startDatetime;
+  @Nullable
+  private final String datetimeAfter;
+  @Nullable
+  private final String datetimeBefore;
+  private final Map<ChronoUnit, Integer> duration;
+  private final Map<ChronoUnit, Integer> offset;
+
+  public SObjectFilterDescriptor(@Nullable String datetimeAfter,
+                                 @Nullable String datetimeBefore,
+                                 Map<ChronoUnit, Integer> duration,
+                                 Map<ChronoUnit, Integer> offset) {
+    this(null, datetimeAfter, datetimeBefore, duration, offset);
+  }
+
+  public SObjectFilterDescriptor(@Nullable Long startTime,
+                                 @Nullable String datetimeAfter,
+                                 @Nullable String datetimeBefore,
+                                 Map<ChronoUnit, Integer> duration,
+                                 Map<ChronoUnit, Integer> offset) {
+    this.startDatetime = toZonedDateTime(startTime);
+    this.datetimeAfter = datetimeAfter;
+    this.datetimeBefore = datetimeBefore;
+    this.duration = duration;
+    this.offset = offset;
+  }
+
+  public ZonedDateTime getStartDatetime() {
+    return startDatetime;
+  }
+
+  @Nullable
+  public String getDatetimeAfter() {
+    return datetimeAfter;
+  }
+
+  @Nullable
+  public String getDatetimeBefore() {
+    return datetimeBefore;
+  }
+
+  public Map<ChronoUnit, Integer> getDuration() {
```

Comment:
Instead of putting the logic in the util class, I think it's better if this class handles the duration/offset calculations and only exposes getDatetimeBefore() and getDatetimeAfter().